File: protocol/lua/ProtoSceneServer.lua
 
Proto4z.register(4000,"SceneKnock") 
Proto4z.SceneKnock = {} --战场服务器挂载 
Proto4z.SceneKnock.__protoID = 4000 
Proto4z.SceneKnock.__protoName = "SceneKnock" 
Proto4z.SceneKnock[1] = {name="lineID", type="ui64" } --分线ID 
Proto4z.SceneKnock[2] = {name="pubHost", type="string" }  
Proto4z.SceneKnock[3] = {name="pubPort", type="ui16" }  
 
Proto4z.register(4001,"ScenePulse") 
Proto4z.ScenePulse = {} --集群脉冲 
Proto4z.ScenePulse.__protoID = 4001 
Proto4z.ScenePulse.__protoName = "ScenePulse" 
 
Proto4z.register(4002,"SceneServerJoinGroupIns") 
Proto4z.SceneServerJoinGroupIns = {} --创建/加入编队, 来自docker的指令 
Proto4z.SceneServerJoinGroupIns.__protoID = 4002 
Proto4z.SceneServerJoinGroupIns.__protoName = "SceneServerJoinGroupIns" 
Proto4z.SceneServerJoinGroupIns[1] = {name="refresh", type="ui64" } --0为客户端请求操作, 1为仅仅刷新数据 
Proto4z.SceneServerJoinGroupIns[2] = {name="groupID", type="ui64" } --编队ID, 0为创建 
Proto4z.SceneServerJoinGroupIns[3] = {name="modelID", type="ui64" } --模型ID 
Proto4z.SceneServerJoinGroupIns[4] = {name="equips", type="DictArrayKey" } --额外装备 
Proto4z.SceneServerJoinGroupIns[5] = {name="avatarID", type="ui64" }  
Proto4z.SceneServerJoinGroupIns[6] = {name="avatarName", type="string" }  
Proto4z.SceneServerJoinGroupIns[7] = {name="rankScore", type="double" } --天梯分数 
 
Proto4z.register(4003,"SceneServerJoinGroupAck") 
Proto4z.SceneServerJoinGroupAck = {} --创建/加入编队响应结果 
Proto4z.SceneServerJoinGroupAck.__protoID = 4003 
Proto4z.SceneServerJoinGroupAck.__protoName = "SceneServerJoinGroupAck" 
Proto4z.SceneServerJoinGroupAck[1] = {name="retCode", type="ui16" }  
Proto4z.SceneServerJoinGroupAck[2] = {name="newGroupID", type="ui64" }  
Proto4z.SceneServerJoinGroupAck[3] = {name="oldGroupID", type="ui64" }  
 
Proto4z.register(4004,"SceneServerEnterSceneIns") 
Proto4z.SceneServerEnterSceneIns = {} --开辟场景 
Proto4z.SceneServerEnterSceneIns.__protoID = 4004 
Proto4z.SceneServerEnterSceneIns.__protoName = "SceneServerEnterSceneIns" 
Proto4z.SceneServerEnterSceneIns[1] = {name="sceneType", type="ui16" } --类型 
Proto4z.SceneServerEnterSceneIns[2] = {name="mapID", type="ui64" }  
Proto4z.SceneServerEnterSceneIns[3] = {name="groups", type="SceneGroupInfoArray" } --多个编队 
 
Proto4z.register(4005,"SceneServerCancelSceneIns") 
Proto4z.SceneServerCancelSceneIns = {} --退出场景 
Proto4z.SceneServerCancelSceneIns.__protoID = 4005 
Proto4z.SceneServerCancelSceneIns.__protoName = "SceneServerCancelSceneIns" 
Proto4z.SceneServerCancelSceneIns[1] = {name="sceneID", type="ui64" } --sceneID 
Proto4z.SceneServerCancelSceneIns[2] = {name="groupID", type="ui64" } --groupID 
 
Proto4z.register(4006,"SceneServerGroupStateFeedback") 
Proto4z.SceneServerGroupStateFeedback = {} --scene ==> world 
Proto4z.SceneServerGroupStateFeedback.__protoID = 4006 
Proto4z.SceneServerGroupStateFeedback.__protoName = "SceneServerGroupStateFeedback" 
Proto4z.SceneServerGroupStateFeedback[1] = {name="sceneID", type="ui64" }  
Proto4z.SceneServerGroupStateFeedback[2] = {name="groupID", type="ui64" }  
Proto4z.SceneServerGroupStateFeedback[3] = {name="state", type="ui16" } --如果是NONE 说明离开场景 
 
Proto4z.register(4008,"EntityScriptNotice") 
Proto4z.EntityScriptNotice = {} --同步给脚本 
Proto4z.EntityScriptNotice.__protoID = 4008 
Proto4z.EntityScriptNotice.__protoName = "EntityScriptNotice" 
Proto4z.EntityScriptNotice[1] = {name="controls", type="EntityControlArray" }  
Proto4z.EntityScriptNotice[2] = {name="skills", type="EntitySkillSystemArray" }  

File: protocol/CSharp/ProtoSceneServer.cs
 
namespace Proto4z  
{ 
 
    public class SceneKnock: Proto4z.IProtoObject //战场服务器挂载  
    {     
        //proto id   
        public const ushort protoID = 4000;  
        static public ushort getProtoID() { return 4000; } 
        static public string getProtoName() { return "SceneKnock"; } 
        //members   
        public ulong lineID; //分线ID  
        public string pubHost;  
        public ushort pubPort;  
        public SceneKnock()  
        { 
            lineID = 0;  
            pubHost = "";  
            pubPort = 0;  
        } 
        public SceneKnock(ulong lineID, string pubHost, ushort pubPort) 
        { 
            this.lineID = lineID; 
            this.pubHost = pubHost; 
            this.pubPort = pubPort; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.lineID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeString(this.pubHost)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI16(this.pubPort)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.lineID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.pubHost = Proto4z.BaseProtoObject.decodeString(binData, ref pos); 
            this.pubPort = Proto4z.BaseProtoObject.decodeUI16(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class ScenePulse: Proto4z.IProtoObject //集群脉冲  
    {     
        //proto id   
        public const ushort protoID = 4001;  
        static public ushort getProtoID() { return 4001; } 
        static public string getProtoName() { return "ScenePulse"; } 
        //members   
        public ScenePulse()  
        { 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            return pos; 
        } 
    } 
 
    public class SceneServerJoinGroupIns: Proto4z.IProtoObject //创建/加入编队, 来自docker的指令  
    {     
        //proto id   
        public const ushort protoID = 4002;  
        static public ushort getProtoID() { return 4002; } 
        static public string getProtoName() { return "SceneServerJoinGroupIns"; } 
        //members   
        public ulong refresh; //0为客户端请求操作, 1为仅仅刷新数据  
        public ulong groupID; //编队ID, 0为创建  
        public ulong modelID; //模型ID  
        public DictArrayKey equips; //额外装备  
        public ulong avatarID;  
        public string avatarName;  
        public double rankScore; //天梯分数  
        public SceneServerJoinGroupIns()  
        { 
            refresh = 0;  
            groupID = 0;  
            modelID = 0;  
            equips = new DictArrayKey();  
            avatarID = 0;  
            avatarName = "";  
            rankScore = 0.0;  
        } 
        public SceneServerJoinGroupIns(ulong refresh, ulong groupID, ulong modelID, DictArrayKey equips, ulong avatarID, string avatarName, double rankScore) 
        { 
            this.refresh = refresh; 
            this.groupID = groupID; 
            this.modelID = modelID; 
            this.equips = equips; 
            this.avatarID = avatarID; 
            this.avatarName = avatarName; 
            this.rankScore = rankScore; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.refresh)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.groupID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.modelID)); 
            if (this.equips == null) this.equips = new DictArrayKey(); 
            data.AddRange(this.equips.__encode()); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.avatarID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeString(this.avatarName)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeDouble(this.rankScore)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.refresh = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.groupID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.modelID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.equips = new DictArrayKey(); 
            this.equips.__decode(binData, ref pos); 
            this.avatarID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.avatarName = Proto4z.BaseProtoObject.decodeString(binData, ref pos); 
            this.rankScore = Proto4z.BaseProtoObject.decodeDouble(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class SceneServerJoinGroupAck: Proto4z.IProtoObject //创建/加入编队响应结果  
    {     
        //proto id   
        public const ushort protoID = 4003;  
        static public ushort getProtoID() { return 4003; } 
        static public string getProtoName() { return "SceneServerJoinGroupAck"; } 
        //members   
        public ushort retCode;  
        public ulong newGroupID;  
        public ulong oldGroupID;  
        public SceneServerJoinGroupAck()  
        { 
            retCode = 0;  
            newGroupID = 0;  
            oldGroupID = 0;  
        } 
        public SceneServerJoinGroupAck(ushort retCode, ulong newGroupID, ulong oldGroupID) 
        { 
            this.retCode = retCode; 
            this.newGroupID = newGroupID; 
            this.oldGroupID = oldGroupID; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI16(this.retCode)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.newGroupID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.oldGroupID)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.retCode = Proto4z.BaseProtoObject.decodeUI16(binData, ref pos); 
            this.newGroupID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.oldGroupID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class SceneServerEnterSceneIns: Proto4z.IProtoObject //开辟场景  
    {     
        //proto id   
        public const ushort protoID = 4004;  
        static public ushort getProtoID() { return 4004; } 
        static public string getProtoName() { return "SceneServerEnterSceneIns"; } 
        //members   
        public ushort sceneType; //类型  
        public ulong mapID;  
        public SceneGroupInfoArray groups; //多个编队  
        public SceneServerEnterSceneIns()  
        { 
            sceneType = 0;  
            mapID = 0;  
            groups = new SceneGroupInfoArray();  
        } 
        public SceneServerEnterSceneIns(ushort sceneType, ulong mapID, SceneGroupInfoArray groups) 
        { 
            this.sceneType = sceneType; 
            this.mapID = mapID; 
            this.groups = groups; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI16(this.sceneType)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.mapID)); 
            if (this.groups == null) this.groups = new SceneGroupInfoArray(); 
            data.AddRange(this.groups.__encode()); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.sceneType = Proto4z.BaseProtoObject.decodeUI16(binData, ref pos); 
            this.mapID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.groups = new SceneGroupInfoArray(); 
            this.groups.__decode(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class SceneServerCancelSceneIns: Proto4z.IProtoObject //退出场景  
    {     
        //proto id   
        public const ushort protoID = 4005;  
        static public ushort getProtoID() { return 4005; } 
        static public string getProtoName() { return "SceneServerCancelSceneIns"; } 
        //members   
        public ulong sceneID; //sceneID  
        public ulong groupID; //groupID  
        public SceneServerCancelSceneIns()  
        { 
            sceneID = 0;  
            groupID = 0;  
        } 
        public SceneServerCancelSceneIns(ulong sceneID, ulong groupID) 
        { 
            this.sceneID = sceneID; 
            this.groupID = groupID; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.sceneID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.groupID)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.sceneID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.groupID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class SceneServerGroupStateChangeIns: Proto4z.IProtoObject //scene ==> world  
    {     
        //proto id   
        public const ushort protoID = 4006;  
        static public ushort getProtoID() { return 4006; } 
        static public string getProtoName() { return "SceneServerGroupStateChangeIns"; } 
        //members   
        public ulong sceneID;  
        public ulong groupID;  
        public ushort state; //如果是NONE 说明离开场景  
        public SceneServerGroupStateChangeIns()  
        { 
            sceneID = 0;  
            groupID = 0;  
            state = 0;  
        } 
        public SceneServerGroupStateChangeIns(ulong sceneID, ulong groupID, ushort state) 
        { 
            this.sceneID = sceneID; 
            this.groupID = groupID; 
            this.state = state; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.sceneID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.groupID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI16(this.state)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.sceneID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.groupID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.state = Proto4z.BaseProtoObject.decodeUI16(binData, ref pos); 
            return pos; 
        } 
    } 
 
} 
 
 
